project1/src/main/java/uk/co/cpsd/javaproject1/World.java

```diff
@@ -55,11 +70,38 @@ public Stream<Animal> animals(){
         return animals.stream();
     }
 
+    //=============================
+
+    public void writeToCSV(List<Integer> goatHistory, List<Integer> grassHistory){
+        
+        try{
+
+            FileWriter csvData=new FileWriter("data.csv");
+            csvData.write("Tick, NumOfGoat, NumOfGrass \n");
+            for(int i=0;i<goatHistory.size();i++){
+                csvData.write(i+", "+goatHistory.get(i)+", "+grassHistory.get(i));
+                csvData.write("\n");
+            }
+            csvData.close();
+
+        }catch(IOException e){
+            System.out.println(e);
+        }
+    }    
+
 
     public void tick() {
         totalTicks++;
         growGrass();
-        
+        growGrass();
+        growGrass();
+        growGrass();
+        growGrass();
+        growGrass();
+        int goatCount = findNumOfGoats();
+        goatHistory.add(goatCount);
+        grassHistory.add(findNumOfGrass());
```

Comment:
You're being inconsistent.  For goats you get a local variable, for grass you don't.  Consistency helps understanding